tests/rptest/services/nessie_catalog.py

```diff
@@ -0,0 +1,197 @@
+        },
+    }
+
+    # The endpoint that directly accesses the iceberg catalog.
+    _iceberg_url: Optional[str] = None
+
+    @property
+    def iceberg_url(self) -> str:
+        assert self._iceberg_url, "URL not available because service is not started"
+        return self._iceberg_url
+
+    @staticmethod
+    def catalog_name() -> str:
+        return NessieCatalog.catalog_type().value
+
+    @staticmethod
+    def catalog_type() -> CatalogType:
+        return CatalogType.NESSIE
+
+    def client(self, catalog_name: Optional[str] = None):
+        return self._client(catalog_name, self._iceberg_url)
+
+    def _java_home(self, node):
+        return node.account.ssh_output(
+            "echo /usr/lib/jvm/java-21-openjdk-$(dpkg-architecture -q DEB_BUILD_ARCH)"
+        ).decode('utf-8').strip()
+
+    def _java_bin(self, node):
+        java_home = self._java_home(node)
+        return f"{java_home}/bin/java"
+
+    def _make_env(self):
+        env = dict()
+        env["NESSIE_CATALOG_DEFAULT_WAREHOUSE"] = self._warehouse_name
+        env[f"NESSIE_CATALOG_WAREHOUSES_{self._warehouse_name.upper()}_LOCATION"] = self.cloud_storage_warehouse
+
+        if isinstance(self.credentials, cloud_storage.S3Credentials):
+            env["NESSIE_CATALOG_SERVICE_S3_DEFAULT_OPTIONS_REGION"] = self.credentials.region
+            env["NESSIE_CATALOG_SERVICE_S3_DEFAULT_OPTIONS_ENDPOINT"] = self.credentials.endpoint
+            env["NESSIE_CATALOG_VALIDATE_SECRETS"] = "true"
+        elif isinstance(self.credentials,
+                        cloud_storage.ABSSharedKeyCredentials):
+            env["NESSIE_CATALOG_SERVICE_ADLS_DEFAULT_OPTIONS_AUTH_TYPE"] = "STORAGE_SHARED_KEY"
+            env["NESSIE_CATALOG_SERVICE_ADLS_DEFAULT_OPTIONS_ACCOUNT_NAME"] = self.credentials.account_name
+            env["NESSIE_CATALOG_SERVICE_ADLS_DEFAULT_OPTIONS_ACCOUNT_SECRET"] = self.credentials.account_key
+        return env
+
+    def _make_java_properties(self):
+        """
+        These options don't work nicely with conversion to env variable format.
+        Specify them as Java -D properties instead.
+        """
+        d_flags = ""
+        d_flags += "-Dnessie.catalog.service.s3.default-options.access-key=urn:nessie-secret:quarkus:my-secrets-default "
+        d_flags += f"-Dmy-secrets-default.name={self.credentials.access_key} "
+        d_flags += f"-Dmy-secrets-default.secret={self.credentials.secret_key} "
+        d_flags += "-Dnessie.catalog.validate-secrets=true"
+        return d_flags
+
+    def _java_cmd(self, node):
+        java_home = self._java_home(node)
+        java_bin = self._java_bin(node)
+        envs = self._make_env()
+        env = " ".join(f"{k}={v}" for k, v in envs.items())
+        d_props = self._make_java_properties()
+        return f"{env} JAVA_HOME={java_home} nohup {java_bin} {d_props} -jar {NessieCatalog.JAR_PATH} \
+        1>> {NessieCatalog.LOG_FILE} 2>> {NessieCatalog.LOG_FILE} & echo $!"
+
+    def __init__(
```

Comment:
nit: mind moving this up a bit so it's the first method defined? It's a bit hard to find nestled in between these private methods